pages/search.tsx

```diff
@@ -21,24 +21,18 @@ interface SearchEmpty {
 
 type SearchState = SearchResponse | SearchEmpty | SearchInit;
 
+function Index(query: {search: string | null, page: number}) {
 
-
-function Index(query: {search: string | null}) {
-
+  const [searchValue, setSearchValue] = useState<string>(query.search ? decodeURI(query.search) : "");
+  const [submittedSearchValue, setSubmittedSearchValue] = useState<string | undefined>(searchValue);
+  const [pageValue, setPageValue] = useState<number>(query.page || 0);
   const [searchState, setSearchState] = useState<SearchState>({
     type: "INIT",
   });
 
-  const [searchValue, setSearchValue] = useState<string>(query.search ? decodeURI(query.search) : "");
-  const [prevSearchState, setPrevSearchState] = useState<string | undefined>(
-    undefined
-  );
-
   useEffect(() => {
-    if (query.search != null) {
-      search();
-    }
-  }, []);
+    search(pageValue);
```

Comment:
Kan du ikke bare bruke pageValue fra state i search, i stedet for å sende inn som param?

Reply:
Godt poeng. Henger nok igjen fra før jeg refaktorerte en del.